django/core/management/templates.py

```diff
@@ -105,10 +105,10 @@ def handle(self, app_or_project, name, target=None, **options):
             if app_or_project == "app":
                 self.validate_name(os.path.basename(top_dir), "directory")
             if not os.path.exists(top_dir):
-                raise CommandError(
-                    "Destination directory '%s' does not "
-                    "exist, please create it first." % top_dir
-                )
+                try:
+                    os.makedirs(top_dir)
+                except OSError as e:
+                    raise CommandError(e)
```

Comment:
The error handling is missing a test

Reply:
> Should I mock `os.makedirs` to raise a specific error to make the test more deterministic?

I've gave this a try

Reply:
Well, that didn't work because the `self.run_django_admin` helper runs the command in a subprocess. As far as I undertsnad, the mock is applied in the current process, so doesn't trigger. I've tried another approach: by setting the parent directory to be read-only.